node/src/actors/chain_manager/mining.rs

```diff
@@ -739,6 +742,12 @@ fn build_block(
             }
         };
 
+        // The value transfer transaction are sorted from highest to lowest fee, so as
+        // soon as the transaction fee is below our minimum fee, we can just stop searching
+        if transaction_fee < minimum_vtt_fee {
+            break;
```

Comment:
Transactions are sorted by fee/weight, so could happen that a transaction with a fee lower than the minimum_vtt_fee was before than other with a bigger fee but much bigger weight

Reply:
Okay, I missed that when going over the code. 

That leaves two options:
1) I still handle `minimum_vtt_fee` as a flat fee and just continue the iteration.
2) I do implement it as a fee / weight as suggested by @tmpolaczyk.

If I go for case 2, I would abstract the `fee-per-weight` away from the miner and say that if you set `minimum_vtt_fee` to 1, that means you want 1 nanoWIT for the most basic VTT (and more for more complex transactions). The most basic VTT has one input one output with a weight of `1*133 + 1*36*10=493`. So in code that would be something like (with proper rounding):
```
if transaction_fee / (weight / 493) < minimum_vtt_fee {
            break;
}
```

What do you think?

Reply:
There is already a `min_vt_weight` variable there, you can use that instead of 493. If minimum_vtt_fee is set in nanowits, then the code should use floating point division to avoid problems with integer division:

```
if f64::from(transaction_fee) / f64::from(weight) < f64::from(minimum_vtt_fee) / f64::from(min_vt_weight) {
            break;
}
```

If this works then go with option 2, otherwise just leave it like it was before and add a warning that this feature is experimental and the node may skip some transactions with higher fee.